mlir/lib/CAPI/IR/Pass.cpp

```diff
@@ -44,8 +44,23 @@ MlirLogicalResult mlirPassManagerRunOnOp(MlirPassManager passManager,
   return wrap(unwrap(passManager)->run(unwrap(op)));
 }
 
-void mlirPassManagerEnableIRPrinting(MlirPassManager passManager) {
-  return unwrap(passManager)->enableIRPrinting();
+void mlirPassManagerEnableIRPrinting(MlirPassManager passManager,
+                                     bool shouldPrintBeforePass,
+                                     bool shouldPrintAfterPass,
+                                     bool printModuleScope,
+                                     bool printAfterOnlyOnChange,
+                                     bool printAfterOnlyOnFailure) {
+  auto shouldPrintBeforeFn = [shouldPrintBeforePass](Pass *, Operation *) {return shouldPrintBeforePass;};
+  auto shouldPrintAfterFn = [shouldPrintAfterPass](Pass *, Operation *) {return shouldPrintAfterPass;};
```

Comment:
Hi! I am so sorry for the delay. 

I am new to the MLIR world, so my understanding might be incorrect. But, I believe useful implementation from `py::function` to `std::function<bool(Pass *, Operation *)>`  will need additional wrappers around the Pass and Operation class. Should we add these changes to this pull request?
 
 


Thank you so much!

Reply:
Wrappers are already available. This can take a `bool (*shouldPrintAfterPass)(MlirPass, MlirOperation, void *userData)`. Python binding side can define two overloads: one that takes a bool and another that takes a `std::function<bool (MlirPass, MlirOperation)>`, both using lambdas to delegate further. I believe we have type casters in place for this to work pretty much out of the box.